HeterogeneousRecyclerView/app/src/main/java/com/charlesdrews/heterogeneousrecyclerview/FootballRecyclerViewAdapter.java

```diff
@@ -0,0 +1,72 @@
+package com.charlesdrews.heterogeneousrecyclerview;
+
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.ViewGroup;
+
+import com.charlesdrews.heterogeneousrecyclerview.model.BaseFootballObject;
+import com.charlesdrews.heterogeneousrecyclerview.model.Player;
+import com.charlesdrews.heterogeneousrecyclerview.model.Team;
+
+import java.util.List;
+
+/**
+ * Created by charlie on 11/14/16.
+ */
+
+public class FootballRecyclerViewAdapter extends RecyclerView.Adapter {
+
+    private List<BaseFootballObject> mFootballObjects;
+
+    public FootballRecyclerViewAdapter(List<BaseFootballObject> footballObjects) {
+        mFootballObjects = footballObjects;
+    }
+
+    @Override
+    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        switch (viewType){
+            case 0:
+                return new PlayerViewHolder(
+                        LayoutInflater.from(parent.getContext())
+                                .inflate(R.layout.player_list_entry,parent,false));
+            case 1:
+                return new TeamViewHolder(
+                        LayoutInflater.from(parent.getContext())
+                                .inflate(R.layout.team_list_entry,parent,false));
+            default:
+                return null;
+        }
+    }
+
+    @Override
+    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
+        if (holder instanceof PlayerViewHolder){
```

Comment:
This works, but I recommend calling `holder.getItemViewType()` instead, which gives you the `int` value assigned to the holder by your overridden `getItemViewType()` below. The `instanceof` operator relies on _reflection_ which is slow, so best not to use it when you don't have to.